lib/resty/core/shdict.lua

```diff
@@ -380,6 +383,16 @@ local function shdict_incr(zone, key, value, init)
 end
 
 
+local function shdict_incr(zone, key, value, init)
+    return shdict_incr_helper(zone, 0x0001, key, value, init)
+end
+
+
+local function shdict_decr(zone, key, value, init)
+    return shdict_incr_helper(zone, 0x0002, key, value, init)
```

Comment:
Can we just reuse the existing C API and negate `value` here? If we want to reject decrementing a negative value, we should do the check just here. But frankly I don't think it's worth it for an extra branch.